src/shape.cpp

```diff
@@ -420,6 +420,20 @@ void shape::multi_copy(std::size_t idx, std::size_t* start, const std::size_t* e
     *start = tidx;
 }
 
+bool shape::check_within_bounds(std::vector<std::size_t> multi) const
+{
+    assert(this->lens().size() == multi.size());
+    bool ret = true;
+    for(int i = 0; i < multi.size(); ++i)
+    {
+        if(multi.at(i) > this->lens().at(i))
```

Comment:
This is a bug, if the index is equal to the lens it will be out of bounds. I think switching to using `std::less<>{}` with `std::equal` will fix this issue.

Reply:
This API implied to me that two shapes were being considered. If instead it is just an indices vector to compare to, the name (of the API) could be reworded. Thanks.

Reply:
Renamed to multi_within_bounds to address Paul and Lakhinder's comments.